doc/compiler/runtime/J9JITExceptionTable.md

```diff
@@ -0,0 +1,400 @@
+<!--
+Copyright IBM Corp. and others 2023
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# Background
+
+The `J9JITExceptionTable` (also `typedef`d to `TR_MethodMetaData`)
+is generally referred to as "the metadata". It contains most of the
+metadata associated with a compiled body (the other metadata
+is stored in the `TR_PersistentJittedBodyInfo`). Most of this metadata
+is stored in the variable length section of the `J9JITExceptionTable`
```

Comment:
Could you elaborate on how we search different variable length sections following the `J9JITExceptionTable` data structure? So if I have a pointer to `J9JITExceptionTable`, how to jump to the sections following `J9JITExceptionTable`? After looking at the fields of `J9JITExceptionTable` data structure, I'm guessing `J9JITExceptionTable` data structure contains pointers to each section? If I have a pointer to `J9JITExceptionTable` for example `ptrExpTbl`, I can access `Stack Atlas` using `ptrExpTbl->gcStackAtlas`, but I don't see pointers for `Stack Alloc Map` or `Internal Pointer Map in J9 Format`.

Reply:
I can look into it, but I don't fully know of all of this works either heh; I know that some of the sections have direct pointers as you mentioned; however, others are probably calculated. The Internal Pointer Map _may_ be pointed to by `J9JITStackAtlas.internalPointerMap`. If it's easy enough for me to figure it all out I'll add the appropriate information.